### ISSUE TITLE
test data

### ISSUE THREAD

**Issue by WAN-L** [pirxus/personalVAD#3]
Hi, thank you for uploading your code!
I want to use the pretrained models to test my own data, but it seems that there is something wrong during the process of feature extraction. So I'd like to ask how should I use the model to test my own data?

**Comment by pirxus** [comment]
Hi!

I have to be honest, I haven't really touched this project for over a year... Could you perhaps be more specific as to where did your experiment fail? Right now, the best way to find out how to use the model properly is to follow the steps for example in the personal_vad_evaluate.ipynb notebook found in the src/ directory. Let me know if you have any luck with this.

One thing that I regret not adding to the project in the end would be a simple API for the model. I guess I should probably do that still, so that issues like these are prevented in the future.

**Comment by WAN-L** [comment]
Hi, I want to use the model to test just one audio, but i don't know how to input it correctly to the model,it seems that  I should get fbanks.scp/labels.scp/scores.scp/targets.scp , so I tried to use "bash prepare_dataset_features.sh 3" to run just stage 3 to get them, right? But there always some errors.

**Comment by WAN-L** [comment]
From the code,I found that "extract_features.py" needs loading the utterance transcriptions at first. Does it means that if I just have an audio without the text, I can't use the model?

**Comment by pirxus** [comment]
Hi,
these scripts you are talking about are meant to be used for creating your own whole training dataset, not for evaluation or inference - only really compatible with the LibriSpeech dataset, as you need individual word alignments for all the utterances in the dataset.

**Comment by pirxus** [comment]
You know what, I will create a simple API for inference later today, so that you don't have to struggle with digging out the whole recipe for it from the codebase.. But beware - as stated in the thesis - you might be disappointed in the results as the models aren't very well tested for real-world data, as it was at the time of writing the thesis really quite difficult to obtain such data.

**Comment by pirxus** [comment]
I'm actually quite interested in how well you will find the models to work - would love some feedback once you get it working :))

**Comment by WAN-L** [comment]
Thank you for your reply! I have tried to use just the part of the original code for extracting embeddings, and  so long as I get these files about features I can use the model. But the result was not as good as I expected.

**Comment by HarshalRaut16** [comment]
> Thank you for your reply! I have tried to use just the part of the original code for extracting embeddings, and so long as I get these files about features I can use the model. But the result was not as good as I expected.

Hello @WAN-L ,
can you please guide me how you had extracted the embeddings and use the model for inference?
